### PR TITLE
Fixes #37774 - wait for the host page to render before editing again

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: fe5c585bd98cb8612d7d401c96b0ed1dd7d5f63d

### CODE REVIEW

File: test/integration/host_js_test.rb [hunk 1]
```diff
@@ -529,6 +529,7 @@ class HostJSTest < IntegrationTestWithJavascript
       page.find(id).click
       assert page.has_no_selector?(id)
       click_button('Submit')
+      find('h5', :text => host.fqdn) # wait for the host page to load
```
Comment (evgeni) [inline test/integration/host_js_test.rb:532]: There is a dedicated `click_on_submit` helper (introduced in 1580adbec7995b3d27e55d4067a2d5a4fd34d291), but all of its uses were removed in 98c17528a655148bf4c01629ea36cf249441c213 and I am not sure why.

Following the same logic as in 98c17528a655148bf4c01629ea36cf249441c213 and using a `find` to find the header with the hostname.

**Comment by evgeni** [comment]
#10298 and #10302 were tested together in https://github.com/theforeman/foreman/pull/10125 and turn up green \o/

File: test/integration/host_js_test.rb (hunk 1 above)
Comment (ekohl) [reply test/integration/host_js_test.rb:532]: Perhaps `assert page.has_link?('Edit')` no longer worked?

Comment (evgeni) [reply test/integration/host_js_test.rb:532]: Idk. Maybe. I just want #10125 to pass :D

**Review by MariaAga — APPROVED** [review]
Also LGTM, thank you!

**Pull request merged**